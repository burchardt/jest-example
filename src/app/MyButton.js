import React from "react";

export class MyButton extends React.Component {
    constructor(props) {
        super(props);
        this.state = {open: true}
    }

    toggleState() {
        this.setState({open : !this.state.open});
    };

    getButtonText() {
       return this.state.open ? "Open" : "Close";
    }

    render() {
        const buttonStyle = {
            "fontSize": "25px",
            "backgroundColor": "lightblue"
        };

        return(
            <div>
                <button id="myButton"
                    className="btn"
                    style={buttonStyle}
                    onClick={this.toggleState.bind(this)}
                >
                {this.getButtonText()}
                </button>
            </div>
        );
    }
}