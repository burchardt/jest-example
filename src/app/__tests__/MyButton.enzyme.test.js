import React from 'react';
import { shallow } from 'enzyme';

import { MyButton } from '../MyButton';

describe('<MyButton />', () => {
    it('should render a button with a classname', () => {
        const wrapper = shallow(
            <MyButton />
        );
        // Generate and match snapshot
        expect(wrapper).toMatchSnapshot();
    });

    it('toggleState should be called on click', () => {
        spyOn(MyButton.prototype, 'toggleState').and.callThrough();
        const wrapper = shallow(
            <MyButton />
        );
        // Find button element and simulate the click
        wrapper.find('#myButton').simulate('click');
        // Check if the toggleState method have been called
        expect(MyButton.prototype.toggleState).toHaveBeenCalled();
        // Generate snapshot after the button was clicked
        expect(wrapper).toMatchSnapshot();
    });

    it('getButtonText should be called', () => {
        spyOn(MyButton.prototype, 'getButtonText').and.returnValue("BUTTON");
        const wrapper = shallow(
            <MyButton />
        );
        // Check if the getButtonText method have been called
        expect(MyButton.prototype.getButtonText).toHaveBeenCalled();
        // Generate and match snapshot
        expect(wrapper).toMatchSnapshot();
    });
});