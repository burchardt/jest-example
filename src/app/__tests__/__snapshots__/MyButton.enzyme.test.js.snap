// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MyButton /> getButtonText should be called 1`] = `
<div>
  <button
    className="btn"
    id="myButton"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "lightblue",
        "fontSize": "25px",
      }
    }
  >
    BUTTON
  </button>
</div>
`;

exports[`<MyButton /> should render a button with a classname 1`] = `
<div>
  <button
    className="btn"
    id="myButton"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "lightblue",
        "fontSize": "25px",
      }
    }
  >
    Open
  </button>
</div>
`;

exports[`<MyButton /> toggleState should be called on click 1`] = `
<div>
  <button
    className="btn"
    id="myButton"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "lightblue",
        "fontSize": "25px",
      }
    }
  >
    Close
  </button>
</div>
`;
